apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/PooledSession.cs

```diff
@@ -0,0 +1,334 @@
+            }
+            return Client.GetSqlStreamReader(request, _session, timeoutSeconds);
+        }
+
+        /// <summary>
+        /// Executes an ExecuteSql RPC asynchronously.
+        /// </summary>
+        /// <param name="request">The query request. Must not be null. The request will be modified with session and transaction details
+        /// from this object. If this object's <see cref="TransactionId"/> is null, the request's transaction is not modified.</param>
+        /// <param name="timeoutSeconds">The timeout for this RPC, in seconds.</param>
+        /// <param name="cancellationToken">An optional token for canceling the call.</param>
+        /// <returns>A task representing the asynchronous operation. When the task completes, the result is the response from the RPC.</returns>
+        public Task<ResultSet> ExecuteSqlAsync(ExecuteSqlRequest request, int timeoutSeconds, CancellationToken cancellationToken)
+        {
+            CheckNotDisposed();
+            GaxPreconditions.CheckNotNull(request, nameof(request));
+            request.SessionAsSessionName = SessionName;
+            if (TransactionId != null)
+            {
+                request.Transaction = new TransactionSelector { Id = TransactionId };
+            }
+            var settings = CreateSettings(Client.Settings.ExecuteSqlSettings, timeoutSeconds, cancellationToken);
+            return RecordSuccessAndExpiredSessions(Client.ExecuteSqlAsync(request, settings));
+        }
+
+        /// <summary>
+        /// Executes a BeginTransaction RPC asynchronously.
+        /// </summary>
+        /// <remarks>
+        /// This method does not affect <see cref="TransactionId"/> of this object.
+        /// </remarks>
+        /// <param name="request">The begin-transaction request. Must not be null. The request will be modified with session details
+        /// from this object.</param>
+        /// <param name="timeoutSeconds">The timeout for this RPC, in seconds.</param>
+        /// <param name="cancellationToken">An optional token for canceling the call.</param>
+        /// <returns>A task representing the asynchronous operation. When the task completes, the result is the response from the RPC.</returns>
+        public Task<Transaction> BeginTransactionAsync(BeginTransactionRequest request, int timeoutSeconds, CancellationToken cancellationToken)
+        {
+            CheckNotDisposed();
+            GaxPreconditions.CheckNotNull(request, nameof(request));
+            request.SessionAsSessionName = SessionName;
+            var settings = CreateSettings(Client.Settings.BeginTransactionSettings, timeoutSeconds, cancellationToken);
+            return RecordSuccessAndExpiredSessions(Client.BeginTransactionAsync(request, settings));
+        }
+
+        private async Task<T> RecordSuccessAndExpiredSessions<T>(Task<T> task, [CallerMemberName] string caller = null)
```

Comment:
I've simplified this method to remove all the logging bits - they were when I was trying to diagnose something else.

Reply:
OK, but you haven't pushed the commit with the simplification yet, right?
Or else maybe remove the `StringBuilder log` as well?